src/main/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImpl.java

```diff
@@ -40,6 +41,44 @@
         return query.getResultList();
     }
 
+    /**
+     * Gets from the data base all the comments, grouped by student, of one section
+     * @param sectionId
+     * @return a map with pairing a student with all his/her comments
+     */
+    @Override
+    public Map<Long, String> getAttendanceCommentsBySectionId(long sectionId) {
+        Validate.notNull(sectionId, "The sectionId parameter must be not null");
+
+        String jpql ="SELECT NEW edu.ksu.canvas.attendance.repository.AttendanceCommentEntry(att.attendanceStudent.studentId, att.dateOfClass, att.notes) " +
+                    "FROM Attendance att " +
+                    "WHERE att.notes IS NOT NULL AND att.attendanceStudent.canvasSectionId = :canvasSectionId ";
+
+        Query query = entityManager.createQuery(jpql);
+        query.setParameter("canvasSectionId", sectionId);
+
+        List<AttendanceCommentEntry> results = (List<AttendanceCommentEntry>) query.getResultList();
+        Map<Long, String> studentCommentsMap = groupCommentsByStudents(results);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "studentCommentsMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1488)